babel/src/board.js

```diff
@@ -283,20 +280,29 @@ export class Board {
         let hasWideStyleMatch = _.some(_.map(flatChunks, hasMatchInSingleRow));
         return hasWideStyleMatch || _.some(_.map(chunks), hasMatchInPairOfRows);
     }
+    
+    hasMatchEvent() {
+        if (findMatches(this.orbs)[0]) { return true };
+    }
 
     swap(swapOrbs) {
-        let [[x1, y1], [x2, y2]] = swapOrbs;
+        let [[row1, col1], [row2, col2]] = swapOrbs;
         let orbsBefore = _.cloneDeep(this.orbs);
-        this.orbs[x1][y1] = orbsBefore[x2][y2]
-        this.orbs[x2][y2] = orbsBefore[x1][y1]
+        this.orbs[row1][col1] = orbsBefore[row2][col2]
+        this.orbs[row2][col2] = orbsBefore[row1][col1]
 
         // undo the swap if it did not yeild a match
-        if (!findMatches(this.orbs)[0]) {
+        if (!this.hasMatchEvent()) {
             this.orbs = orbsBefore;
         };
     }
 
-    shuffle() {
-
```

Comment:
Okay. I currently have `createValidOrbs()` which will make a whole new board. But I can see why it needs a separate `shuffle()` function. If the board evaluates and now `hasMatch()` is false, it `needsShuffle()`. Would it be more ideal to have evaluate make sure it drops in orbs that will ensure the board `hasMatch`?